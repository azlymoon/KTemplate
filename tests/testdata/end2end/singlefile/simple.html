{{ 10 }}
{% let $null = null %}
{{ $null }}
{% let $s1 = 'aaa' %}
{{ $s1 }}
test_name='{{ test_name }}'
{% let $bool_true = true %}
true->{{ $bool_true }}
{% let $bool_false = false %}
false->{{ $bool_false }}
