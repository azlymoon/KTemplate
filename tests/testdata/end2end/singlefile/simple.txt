{{ 10 }}
{% let $null = null %}
{{ $null }}
{% let $s1 = 'aaa' %}
{{ $s1 }}
test_name='{{ test_name }}'
{% let $bool_true = true %}
true->{{ $bool_true }}
{% let $bool_false = false %}
false->{{ $bool_false }}
!true->{{ not $bool_true }}
!false->{{ not $bool_false }}
!!true->{{ not not $bool_true }}
!!false->{{ not not $bool_false }}
!!!true->{{ not not not $bool_true }}
!!!false->{{ not not not $bool_false }}
$s1 == 'aaa'->{{ $s1 == 'aaa' }}
$s1 == ''->{{ $s1 == '' }}
$s1 != 'aaa'->{{ $s1 != 'aaa' }}
$s1 != ''->{{ $s1 != '' }}
$s1 . $s->{{ $s1 ~ $s1 }}
$s1 . $s . $s->{{ $s1 ~ $s1 ~ $s1 }}
{% let $i1 = 1 %}{% let $i2 = 2 %}
$i1 > $i2->{{ $i1 > $i2 }}
$i1 < $i2->{{ $i1 < $i2 }}
$i2 > $i1->{{ $i2 > $i1 }}
$i2 < $i1->{{ $i2 < $i1 }}
{{ time.year }} {{ time.month }}
{{ time.year ~ ':' ~ time.month }}
{{ time.year }} {{ time.month }}
{{ time.year ~ ':' ~ time.month }}
{{ x.y.z }}/{{ x.y.z + x.y.z }}
{{ x.y.z + x.y.z }}/{{ x.y.z }}
{{ (x.y.z) }}-{{ (x).y.z }}-{{ (x).(y).(z) }}
{{ ($i2 + $i2) * 6 }}
{{ $i2 + ($i2 * 6) }}
{{ 'a'|strlen }}
{{ ($s1 ~ $s1)|strlen }}
{{ $s1|strlen + 1 }}
{{ 1|add1 }}
{{ 1|add1|add1 }}
{{ 1|add1|add1|add1 }}
{{ 1|sub1|add1 }}
{{ 1|add1|sub1|add1 }}
{{ 1|sub1|sub1|sub1 }}
{{ test_name|strlen }}
{{ test_name|strlen|add1 }}
{{ 'aaaa'|length }}
{{ x.y|length }}
{{ $i1|add(1) }}
{{ 1|add(10)|add1 }}
{{ $i1|add($i2) }}
{{ zero() }}
{{ abs(zero()) }}
{{ abs(zero() - 1) }}
{{ abs(1) }}
{{ fmt('%b', 255) }}
{{ fmt('%b', zero() + 10) }}
{{ concat3('a', 'b', 'c') }}
{{ fmt(concat3('', '%', 'b'), 8) }}
{{ abs(-1) }}
{{ $i1 - 10 - 20 }}
{{ $i1 + 10 + 20 }}
{{ $i1 * 10 * 20 }}
{{ ($i1+1) * 10 * 20 }}
{{ assert_true($i1 or $i2) }}
{{ assert_true($null or $null) }}
{{ assert_true($null or $bool_false) }}
{{ assert_false($i1 or $i2) }}
{{ assert_false($null or $null) }}
{{ assert_false($null or $bool_false) }}
{{ assert_true(0 or $bool_false or $i1) }}
{{ assert_true($bool_false or 0 or $i1) }}
{{ assert_false($bool_false and 1) }}
{{ assert_false(1 and $bool_false) }}
{{ assert_true(1 and 2) }}
{{ assert_true($bool_false and 1) }}
{{ assert_true(1 and $bool_false) }}
{{ assert_false(1 and 2) }}
{{ assert_true($i1 and $i1) }}
{{ assert_true($i1 and $bool_false) }}
{% let $myvar = 10 %}
{{ $myvar }}
{% set $myvar = 20 %}
{{ $myvar }}
{{ 1-1 }}
{{ $myvar-1 }}
{{ x["y"]["z"] }}
{{ x[0] }}
{{ x["y"][1] }}
{{ x.y[$i1] }}
{{ x[$i1-1] }}
{{ x["y"][x["key"]] }}
{{ $s1[0] }}
{{ ($s1[0] ~ 'b')[1] }}
{% let $arr = arr %}
{{ $arr["x"]["y"]["z"] }}
{{ '\'\x41' }}
{{ "\"" }}
{{ '\n\041\41' }}
{{ "おい" }} == おい
{{ 'привет' }} == привет
