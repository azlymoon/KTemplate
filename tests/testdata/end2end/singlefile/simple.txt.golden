10



aaa
test_name='simple.txt'

true->1

false->
!true->
!false->1
!!true->1
!!false->
!!!true->
!!!false->1
$s1 == 'aaa'->1
$s1 == ''->
$s1 != 'aaa'->
$s1 != ''->1
$s1 . $s->aaaaaa
$s1 . $s . $s->aaaaaaaaa

$i1 > $i2->
$i1 < $i2->1
$i2 > $i1->1
$i2 < $i1->
2022 August
2022:August
2022 August
2022:August
111/222
222/111
111-111-111
24
14
1
6
4
2
3
4
1
2
-2
10
11
4
2
2
12
3
0
0
1
1
11111111
1010
abc
1000
1
-29
31
200
400
yes
no
no
no
yes
yes
yes
yes
yes
yes
yes
no
no
no
yes
no

10

20
0
19
111
zero
one
one
zero
111
a
b

111
'A
"

!!
おい == おい
привет == привет
